### PR TITLE
Tweak crossgen2 test compilation modes

### LINKED ISSUE #38239: CG2: m_alignpad == 0 in readytorun/crossboundarylayout/crossboundarytest (Ubuntu x64)
Example run:

https://dev.azure.com/dnceng/public/_build/results?buildId=698082&view=ms.vss-test-web.build-test-results-tab&runId=21631922&resultId=108010&paneView=attachments

Platform:

Linux x64 (Ubuntu 18.04)

Item log:

<pre>
irm: cannot remove '/home/helixbot/work/C40809FE/w/A0C208F8/e/readytorun/crossboundarylayout/crossboundarytest/crossboundarytest/crossboundarytest.dll.rsp': No such file or directory

Assert failure(PID 7669 [0x00001df5], Thread: 7669 [0x1df5]): m_alignpad == 0
    File: /__w/1/s/src/coreclr/src/vm/syncblk.cpp Line: 2967
    Image: /home/helixbot/work/C40809FE/p/corerun

/home/helixbot/work/C40809FE/w/A0C208F8/e/readytorun/crossboundarylayout/crossboundarytest/crossboundarytest/crossboundarytest.sh: line 317:  7669 Aborted                 $LAUNCHER $ExePath "${CLRTestExecutionArguments[@]}"

Return code:      1
Raw output file:      /home/helixbot/work/C40809FE/w/A0C208F8/e/readytorun/crossboundarylayout/Reports/readytorun.crossboundarylayout/crossboundarytest/crossboundarytest/crossboundarytest.output.txt
Raw output:
BEGIN EXECUTION
in takeLock
Response file: /home/helixbot/work/C40809FE/w/A0C208F8/e/readytorun/crossboundarylayout/crossboundarytest/crossboundarytest/crossboundarytest.dll.rsp
--inputbubble
/home/helixbot/work/C40809FE/w/A0C208F8/e/readytorun/crossboundarylayout/crossboundarytest/crossboundarytest/IL/crossboundarytest.dll
-r:/home/helixbot/work/C40809FE/w/A0C208F8/e/readytorun/crossboundarylayout/crossboundarytest/crossboundarytest/IL/*.dll
-o:/home/helixbot/work/C40809FE/w/A0C208F8/e/readytorun/crossboundarylayout/crossboundarytest/crossboundarytest/crossboundarytest.dll
-r:/home/helixbot/work/C40809FE/p/System.*.dll
-r:/home/helixbot/work/C40809FE/p/Microsoft.*.dll
-r:/home/helixbot/work/C40809FE/p/mscorlib.dll
--targetarch:x64
-O
Running CrossGen2:  dotnet /home/helixbot/work/C40809FE/p/crossgen2/crossgen2.dll @/home/helixbot/work/C40809FE/w/A0C208F8/e/readytorun/crossboundarylayout/crossboundarytest/crossboundarytest/crossboundarytest.dll.rsp 
Emitting R2R PE file: /home/helixbot/work/C40809FE/w/A0C208F8/e/readytorun/crossboundarylayout/crossboundarytest/crossboundarytest/crossboundarytest.dll
in ReleaseLock
/home/helixbot/work/C40809FE/p/corerun crossboundarytest.dll ''
Writing minidump to file /home/helixbot/dotnetbuild/dumps/coredump.7669.dmp
Written 52629504 bytes (12849 pages) to core file
Dump successfully written
Expected: 100
Actual: 134
END EXECUTION - FAILED
Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=/home/helixbot/work/C40809FE/p
> /home/helixbot/work/C40809FE/w/A0C208F8/e/readytorun/crossboundarylayout/crossboundarytest/crossboundarytest/crossboundarytest.sh
</pre>

### PR DESCRIPTION
**Pull request by davidwrighton**
- Change crossgen2 test coverage to compile every assembly in the application, not just the primary assembly
- Disable by default compilation with the --inputbubble flag. Instead, only use it if LargeVersionBubble is set to 1
- Add a specific error return code from crossgen2 for the case where all input assemblies are determined to be native binaries. This is used to support this scripting change

Fixes #38239

### CODE REVIEW

**Comment by davidwrighton** [comment]
@dotnet/crossgen-contrib

File: src/coreclr/tests/src/CLRTest.CrossGen.targets [hunk 1]
```diff
@@ -72,55 +72,68 @@ if [ ! -z ${RunCrossGen2+x} ]%3B then
...
-      echo --inputbubble>>$__ResponseFile
-      echo $PWD/IL/$(MSBuildProjectName).dll>>$__ResponseFile
-      echo -r:$PWD/IL/*.dll>>$__ResponseFile
+        ExtraCrossGen2Args+= -r:$PWD/IL/*.dll
+      for dllFile in $PWD/IL/*.dll
+      do
+        echo $dllFile
+        bareFileName="${dllFile##*/}"
+        OneFileCrossgen2 "$PWD/$bareFileName" "$dllFile"
+        if [ $__cg2ExitCode -ne 0 ]; then
```
Comment (trylek) [inline src/coreclr/tests/src/CLRTest.CrossGen.targets:124]: Should we be tolerant towards the exit code 2 you introduced for non-loadable inputs? At least my understanding is that the new exit code is supposed to cater for the cases where we blindly submit a native DLL to CG2.

File: src/coreclr/tests/src/CLRTest.CrossGen.targets [hunk 2]
```diff
@@ -161,24 +174,41 @@ if defined RunCrossGen (
...
+        IF NOT !CrossGen2Status!==0 goto :DoneCrossgen2Operations
     ) else (
-        set __OutputFile=!scriptPath!\$(MSBuildProjectName).dll
+        for %%I in (!scriptPath!\*.dll) do (
+            set ExtraCrossGen2Args=!ExtraCrossGen2Args! -r:!scriptPath!IL\*.dll
+            set __OutputFile=!scriptPath!\%%~nI.dll
+            set __InputFile=!scriptPath!IL\%%~nI.dll
+            call :CompileOneFileCrossgen2
+            IF NOT !CrossGen2Status!==0 (
+              IF NOT !CrossGen2Status!==2 goto :DoneCrossgen2Operations
```
Comment (trylek) [inline src/coreclr/tests/src/CLRTest.CrossGen.targets:203]: I see the handling of 2 here but not in the bash version.

**Review by trylek — APPROVED** [review]
Looks great overall, thanks for improving this David; just please double-check the Unix version w.r.t. the exit code 2.

File: src/coreclr/tests/src/CLRTest.CrossGen.targets (hunk 1 above)
Comment (davidwrighton) [reply src/coreclr/tests/src/CLRTest.CrossGen.targets:124]: Eh, on Unix there aren't any files with the .dll extension that should have this characteristic.

File: src/coreclr/tests/src/CLRTest.CrossGen.targets (hunk 2 above)
Comment (davidwrighton) [reply src/coreclr/tests/src/CLRTest.CrossGen.targets:203]: On Unix there aren't any files with the .dll extension other than managed things.

File: src/coreclr/tests/src/CLRTest.CrossGen.targets (hunk 1 above)
Comment (trylek) [reply src/coreclr/tests/src/CLRTest.CrossGen.targets:124]: I see, I didn't realize that native libraries would get automatically filtered out on the other OS'es due to different extensions. Thanks for clarifying!

**Pull request merged**